### ISSUE TITLE
Pass data to before_patch and before_post

### ISSUE THREAD

**Issue by shubham-padia** [miLibris/flask-rest-jsonapi#43]
Currently, received data (`json_data` from http://flask-restless.readthedocs.io/en/stable/customizing.html#request-preprocessors-and-postprocessors)
 is not passed to before_patch and before_post, thus modifying the incoming data is not a possibility.
`flask-restlless` currently supports this,
```
def post_preprocessor(data=None, **kw):
    """Accepts a single argument, `data`, which is the dictionary of
    fields to set on the new instance of the model.

    """
    pass
```
Ref: http://flask-restless.readthedocs.io/en/stable/customizing.html#request-preprocessors-and-postprocessors

**Comment by akira-dev** [comment]
Yes you're right. I will make an update asap. Thanks

**Comment by AleksandrShepelev** [comment]
We need this feature too.

**Comment by akira-dev** [comment]
Fixed in f2cb7e94e0452a2f4b59ebea3270b5db6f9ab5ce

**Comment by shubham-padia** [comment]
@akira-dev Thank You :)

**Comment by akira-dev** [comment]
New release of Flask-REST-JSONAPI is available on Pypi 0.12.6
